### DISCUSSION TITLE
Inverse problem for fitting a light curve of single rotate star with spot

### DISCUSSION THREAD

**Discussion by andrewbooks** [phoebe-project/phoebe2#612] [Questions]
Dear Phoebe team,

I recently have a need for fitting an observed light curve, which is most likely produced by a single rotate star with spot.  
        
Now I can use phoebe to produce fake light curves and mesh figures with different spot settings. But I can't find a good way to solve the inverse problem, which is to fit the light curve and obtain the fitted spot parameters, since lc_geometry and ebai don't support single stars. 
My problem, to be exact, is what solver should I use to achieve this goal? Could you point me to the right direction, please? A complete example would be perfect.

Here is a code I used from the example to create fake light curve of a single rotate star with spot:

`import phoebe`
`import numpy as np`

`logger = phoebe.logger('error')`
`np.random.seed(123456789)`

`b = phoebe.default_star()`
`b.add_spot(radius=30, colat=90, long=90, relteff=0.9)`
`b.set_value('teff', 2800)`
`b.set_value('mass', 0.21)`
`b.set_value('requiv', 0.15)`
`b.set_value('period', 0.14)`

`lctimes = phoebe.linspace(0, 2, 500)`
`b.add_dataset('lc', compute_times=lctimes)`
`b.set_value('atm', 'phoenix')`
`b.set_value('ld_mode', 'interp')`
`b.run_compute(irrad_method='none', distortion_method='rotstar')`

`fluxes = b.get_value('fluxes@model') + np.random.normal(size=lctimes.shape) * 0.01`
`fsigmas = np.ones_like(lctimes) * 0.02`
`plt.plot(lctimes,fluxes)`

Thanks in advance!
Andrew

**Comment by kecnry** [comment]
Hi Andrew - 

Spots are notorious for degeneracies (you could get essentially the same light curve from one large spot vs several smaller spots, or by changing the temperature of both the star and spot, etc), so I wouldn't necessarily trust any results you get as suggesting that is actually what is happening on a real star.

That said... you're right, there are currently no estimators for single stars, let alone spots, but all optimizers and samplers should work just as they do for binaries.  You'll need to determine your number of spots manually and try to tweak the parameters to roughly reproduce the observations, then optimize to find the local minimum.  You can sample to determine "posteriors", but again, keep in mind your uncertainties will then assume that you've found the correct solution already and ignore any uncertainties that there are actual additional spots, etc.

Hopefully this at least helps you get started - let us know if you run into any issues/questions!

-Kyle
